### PR TITLE
add explicit types to implicit methods and values

### PR DESCRIPTION
**Pull request by fwbrasil**
#Problem

The lack of explicit types for implicits makes it harder to follow the scalding code.

#Solution

Add explicit types using [scalafix](https://github.com/scalacenter/scalafix).

### CODE REVIEW

File: scalding-commons/src/main/scala/com/twitter/scalding/commons/source/LzoGenericScheme.scala [hunk 1]
```diff
@@ -41,7 +41,7 @@ private[source] object ExternalizerSerializer {
     import com.twitter.bijection.Inversion.attemptWhen
     import com.twitter.bijection.codec.Base64
 
-    implicit val baseInj = JavaSerializationInjection[Externalizer[T]]
+    implicit val baseInj: JavaSerializationInjection[Externalizer[T]] = JavaSerializationInjection[Externalizer[T]]
```
Comment (johnynek) [inline scalding-commons/src/main/scala/com/twitter/scalding/commons/source/LzoGenericScheme.scala]: I think this is really `Injection[Externalizer[T], Array[Byte]]`.

File: scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala [hunk 2]
```diff
@@ -30,7 +30,7 @@ import com.twitter.algebird._
  * than 2 entries.
  */
 object CumulativeSum {
-  implicit def toCumulativeSum[K, U, V](pipe: TypedPipe[(K, (U, V))]) =
+  implicit def toCumulativeSum[K, U, V](pipe: TypedPipe[(K, (U, V))]): CumulativeSum.CumulativeSumExtension[K, U, V] =
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/CumulativeSum.scala]: I don't think we need `CumulativeSum.` here.

File: scalding-core/src/main/scala/com/twitter/scalding/FieldConversions.scala [hunk 3]
```diff
@@ -111,22 +111,22 @@ trait FieldConversions extends LowPriorityFieldConversions {
...
+  implicit def intToFields(x: Int): Fields = new Fields(new java.lang.Integer(x))
+  implicit def integerToFields(x: java.lang.Integer): Fields = new Fields(x)
+  implicit def stringToFields(x: String): Fields = new Fields(x)
+  implicit def enumValueToFields(x: Enumeration#Value): Fields = new Fields(x.toString)
   /**
    * '* means Fields.ALL, otherwise we take the .name
    */
-  implicit def symbolToFields(x: Symbol) = {
+  implicit def symbolToFields(x: Symbol): Fields = {
     if (x == '*) {
```
Comment (fwbrasil) [inline scalding-core/src/main/scala/com/twitter/scalding/FieldConversions.scala:123]: seems to be github's syntax highlighter having problems to render scala symbols

File: scalding-core/src/main/scala/com/twitter/scalding/mathematics/SizeHint.scala [hunk 4]
```diff
@@ -16,7 +16,7 @@ limitations under the License.
 package com.twitter.scalding.mathematics
 
 object SizeHint {
-  implicit val ordering = SizeHintOrdering
+  implicit val ordering: SizeHintOrdering.type = SizeHintOrdering
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/mathematics/SizeHint.scala]: I think this is `Ordering[SizeHint]`

File: scalding-core/src/main/scala/com/twitter/scalding/serialization/WrappedSerialization.scala [hunk 5]
```diff
@@ -87,7 +87,7 @@ object WrappedSerialization {
   type ClassSerialization[T] = (Class[T], Serialization[T])
 
   private def getSerializer[U]: Injection[Externalizer[U], String] = {
-    implicit val initialInj = JavaSerializationInjection[Externalizer[U]]
+    implicit val initialInj: JavaSerializationInjection[Externalizer[U]] = JavaSerializationInjection[Externalizer[U]]
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/serialization/WrappedSerialization.scala]: I think this is `Injection[Externalizer[U], Array[Byte]]`

File: scalding-core/src/main/scala/com/twitter/scalding/typed/Sketched.scala [hunk 6]
```diff
@@ -40,10 +40,10 @@ case class Sketched[K, V](pipe: TypedPipe[(K, V)],
 
   def reducers = Some(numReducers)
 
-  private lazy implicit val cms = CMS.monoid[Bytes](eps, delta, seed)
+  private lazy implicit val cms: com.twitter.algebird.CMSMonoid[Bytes] = CMS.monoid[Bytes](eps, delta, seed)
```
Comment (johnynek) [inline scalding-core/src/main/scala/com/twitter/scalding/typed/Sketched.scala]: probably import these.

File: scalding-hadoop-test/src/test/scala/com/twitter/scalding/platform/PlatformTest.scala [hunk 7]
```diff
@@ -110,8 +110,8 @@ object MultipleGroupByJobData {
 class MultipleGroupByJob(args: Args) extends Job(args) {
   import com.twitter.scalding.serialization._
   import MultipleGroupByJobData._
-  implicit val stringOrdSer = new StringOrderedSerialization()
-  implicit val stringTup2OrdSer = new OrderedSerialization2(stringOrdSer, stringOrdSer)
+  implicit val stringOrdSer: StringOrderedSerialization = new StringOrderedSerialization()
```
Comment (johnynek) [inline scalding-hadoop-test/src/test/scala/com/twitter/scalding/platform/PlatformTest.scala]: OrderedSerialization[String]

File: scalding-hadoop-test/src/test/scala/com/twitter/scalding/platform/PlatformTest.scala [hunk 8]
```diff
@@ -110,8 +110,8 @@ object MultipleGroupByJobData {
...
+  implicit val stringTup2OrdSer: OrderedSerialization2[String, String] = new OrderedSerialization2(stringOrdSer, stringOrdSer)
```
Comment (johnynek) [inline scalding-hadoop-test/src/test/scala/com/twitter/scalding/platform/PlatformTest.scala]: `OrderedSerialization[(String, String)]`

**Review by johnynek — CHANGES_REQUESTED** [review]
some narrowing of types requested.

**Review by johnynek — APPROVED** [review]

**Comment by johnynek** [comment]
👍 

Thanks for doing this. Nice to get some experience with this tool.

You might also try running it on algebird.

**Pull request merged**